Header/graph_color.h

```diff
@@ -0,0 +1,69 @@
+#ifndef _GRAPH_COLOR_H_
+#define _GRAPH_COLOR_H_
+
+#include <map>
+#include <stack>
+#include <locale>
+#include <vector>
+#include <string>
+#include <utility>
+#include <iostream>
+#include <queue>
+
+using std::queue;
+using std::map;
+using std::pair;
+using std::stack;
+using std::string;
+using std::vector;
+using std::cout;
+using std::cerr;
+using std::endl;
+
+namespace GraphColoring {
+    enum Algorithm {kNone,kDSATUR,kMCS,kTABUCOL,kHybrid,kLMXRLF,kHybridDSATUR};
+    class GraphColor {
+        protected:
+           Algorithm algorithm;
+           map<string,vector<string> > graph;
+           map<string,int> coloring;
+           bool colored;
+
+           /* Helper Functions */
+           /* Writing helper function */
+           string get_color_string(int color,int max_color);
+           int find_max_color();
+        public:
+            //Default coloring algorithm is DSATUR
+            GraphColor(map<string, vector<string> > input_graph):colored(false){ 
+              graph = input_graph;
+              algorithm = kNone; 
+            }
+
+            /* Mutators */
+            void add_edge(string source,string sink);
+            void add_node(string new_node) { graph.insert(pair<string,vector<string> >(new_node,vector<string>())); }
+            void set_algorithm(Algorithm new_algorithm) { algorithm = new_algorithm; }
+
+            /* Coloring functions */
+            virtual map<string,int> color(int condition = 0) = 0;
```

Comment:
Remove the condition here, any subclasses that require a condition should have an additional constructor added that can take a condition value